src/rgw/rgw_http_client.cc

```diff
@@ -418,6 +418,16 @@ int RGWHTTPTransceiver::send_data(void* ptr, size_t len)
 }
 
 
+static int clear_signal(int fd)
+{
+  uint32_t buf;
+  int ret = ::read(fd, &buf, sizeof(buf));
```

Comment:
@cbodley need to cast to void * like we had originally? might be that needed for clearing a warning